src/aiida_quantumespresso/calculations/__init__.py

```diff
@@ -709,6 +717,17 @@ def _generate_PWCPinputdata(cls, parameters, settings, pseudos, structure, kpoin
             kpoints_card = ''.join(kpoints_card_list)
             del kpoints_card_list
 
+        # ============ I prepare the CONSTRAINTS card =============
+        constraints = settings.pop('CONSTRAINTS', None)
+        if constraints is not None:
+            input_params['IONS']['ion_dynamics'] = 'fire'
```

Comment:
Is this necessary? If so, I would validate this in `validate_inputs`. We can also set this there if it's more user-friendly, and raise a warning.